src/scheduler/check.c

```diff
@@ -1989,7 +1989,12 @@ void get_resresv_spec(resource_resv *resresv, selspec **spec, place **pl)
 			*spec = resresv->select;
 		}
 	} else if (resresv->is_resv && resresv->resv != NULL) {
-		if (resresv->resv->resv_state == RESV_BEING_ALTERED || resresv->resv->resv_state == RESV_RUNNING)
+		/* The execselect should be used when the resv is running.  We can't
+		 * trust the state/substate to be RESV_RUNNING when a reservation is both
+		 * RESV_DEGRADED and RESV_BEING_ALTERED and is running.
+		 */
+		if (resresv->resv->is_running)
```

Comment:
should this also check RESV_BEING_ALTERED?

Reply:
No, we need to use the execselect only when the reservation is running.  It can be in BEING_ALTERED when it is confirmed.  In that case, the reservation can be confirmed anywhere.  There are 3 cases where we need to use the execselect (all with the reservation running):
1) state = RESV_RUNNING substate = RESV_DEGRADED
2) state = BEING_ALTERED substate = RESV_RUNNING
3) state = BEING_ALTERED substate = RESV_DEGRDED while the reservation is running.  The state/substate will be the same if the reservation is running or not.  I needed to come up with another way to determine if the reservation was running than its state/substate.

Reply:
Oh, this is the actual reason for the PR. I see now.